projects/baby_driver/test/test_i2c_read.c

```diff
@@ -0,0 +1,213 @@
+#include "i2c_read.h"
+
+#include <string.h>
+#include <time.h>
+
+#include "babydriver_msg_defs.h"
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "delay.h"
+#include "dispatcher.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "ms_test_helper_can.h"
+#include "ms_test_helpers.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define I2C_READ_SOFT_TIMER_TIMEOUT_MS 5
+
+typedef enum {
+  TEST_CAN_EVENT_TX = 0,
+  TEST_CAN_EVENT_RX,
+  TEST_CAN_EVENT_FAULT,
+  NUM_TEST_CAN_EVENTS,
+} TestCanEvent;
+
+// Stores data of i2c_read and i2c_read_reg
+static I2CReadCommand s_test_storage = { 0 };
+
+static CanStorage s_can_storage;
+static uint8_t s_received_status;
+static uint8_t s_rx_data_test[255] = { 0 };
+static size_t s_bytes;
+
+StatusCode TEST_MOCK(i2c_read)(I2CPort i2c, I2CAddress address, uint8_t *rx_data, size_t rx_len) {
+  s_test_storage.port = i2c;
+  s_test_storage.address = address;
+  // Make register 0 to indicate it is not a register read
+  s_test_storage.reg = 0;
+  s_test_storage.rx_len = rx_len;
+  memcpy(s_rx_data_test, rx_data, 255);
+  return STATUS_CODE_OK;
+}
+
+StatusCode TEST_MOCK(i2c_read_reg)(I2CPort i2c, I2CAddress address, uint8_t reg, uint8_t *rx_data,
+                                   size_t rx_len) {
+  s_test_storage.port = i2c;
+  s_test_storage.address = address;
+  s_test_storage.reg = reg;
+  s_test_storage.rx_len = rx_len;
+  memcpy(s_rx_data_test, rx_data, 255);
+  return STATUS_CODE_OK;
+}
+
+static StatusCode prv_rx_i2c_read_callback(uint8_t data[8], void *context, bool *tx_result) {
+  for (size_t i = 0; i < 8; i++) {
+    s_rx_data_test[s_bytes] = data[i];
+    if (data[i] != 0) s_bytes++;
+  }
+  *tx_result = false;
+
+  return STATUS_CODE_OK;
+}
+
+void setup_test(void) {
+  initialize_can_and_dependencies(&s_can_storage, SYSTEM_CAN_DEVICE_BABYDRIVER, TEST_CAN_EVENT_TX,
+                                  TEST_CAN_EVENT_RX, TEST_CAN_EVENT_FAULT);
+  TEST_ASSERT_OK(dispatcher_init());
+  dispatcher_register_callback(BABYDRIVER_MESSAGE_I2C_READ_DATA, prv_rx_i2c_read_callback, NULL);
+  i2c_read_init(I2C_READ_SOFT_TIMER_TIMEOUT_MS);
```

Comment:
It looks like you didn't update this when you changed the name of the `I2C_READ_SOFT_TIMER_TIMEOUT_MS` macro, please make sure this test file compiles and passes with `make test PLATFORM=x86 PROJECT=baby_driver TEST=i2c_read` and then I'll look at it further.